tokio-macros/src/entry.rs

```diff
@@ -174,13 +174,29 @@ fn parse_bool(bool: syn::Lit, span: Span, field: &str) -> Result<bool, syn::Erro
 }
 
 fn parse_knobs(
-    mut input: syn::ItemFn,
+    input_tokens: TokenStream,
     args: syn::AttributeArgs,
     is_test: bool,
     rt_multi_thread: bool,
 ) -> Result<TokenStream, syn::Error> {
+    let mut input: syn::ItemFn = syn::parse(input_tokens.clone())?;
+
+    if (is_test || input.sig.ident == "main") && !input.sig.inputs.is_empty() {
+        let function = if is_test { "test" } else { "main" };
+
+        return Err(syn::Error::new_spanned(
+            &input.sig.inputs,
+            format_args!("the {} function cannot accept arguments", function),
+        ));
+    }
```

Comment:
Things annotated with `#[tokio::main]` can accept arguments.

Reply:
This just replicates the behaviour of the old code, which allowed all `#[tokio::main]` functions to accept arguments apart from those named `main`. It is a bit weird though - should I change it?

Reply:
Ah, if the behavior is the same as before, then it is ok.